# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/DecoderTypes.h
# Commit: 2e334c8a1160
# Full Hash: 2e334c8a11602d2783fd894a237f170267b1fa94
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 3
# Description:
#   Bug 1749047 - Allow printing a VideoDecoderConfig and a VideoDecoderConfigInternal. r=chunmin
#   
#   This is very useful to understand in the log what config is output in what
#   encoder output callback, with respect to configure(...) calls.
#   
# ==============================================================================

diff -r be191f9ac293 -r 2e334c8a1160 dom/media/webcodecs/DecoderTypes.h
--- a/dom/media/webcodecs/DecoderTypes.h	Fri Dec 22 21:39:13 2023 +0000
+++ b/dom/media/webcodecs/DecoderTypes.h	Fri Dec 22 21:39:14 2023 +0000
@@ -7,6 +7,7 @@
 #ifndef mozilla_dom_DecoderTypes_h
 #define mozilla_dom_DecoderTypes_h
 
+#include "MediaData.h"
 #include "mozilla/Maybe.h"
 #include "mozilla/dom/EncodedVideoChunk.h"
 #include "mozilla/dom/VideoColorSpaceBinding.h"
@@ -44,6 +45,7 @@
       const VideoDecoderConfig& aConfig);
   ~VideoDecoderConfigInternal() = default;
 
+  nsString ToString();
 
   bool Equals(const VideoDecoderConfigInternal& aOther) const {
     if (mDescription.isSome() != aOther.mDescription.isSome()) {